docs/features/interceptors.md

```diff
@@ -191,7 +191,12 @@ Interceptors are treated like a post-compilation step in this design. Diagnostic
 
 ### User opt-in
 
-Interceptors will require a feature flag during the experimental phase. The flag can be enabled with `/features=InterceptorsPreview` on the command line or `<Features>InterceptorsPreview</Features>` in msbuild.
+To use interceptors, the user project must specify the property `<InterceptorsPreviewNamespaces>`. This is a list of namespaces which are allowed to contain interceptors.
+```xml
+<InterceptorsPreviewNamespaces>$(InterceptorsPreviewNamespaces);Microsoft.AspNetCore.Http.Generated;MyLibrary.Generated</InterceptorsPreviewNamespaces>
+```
+
+It's expected that each entry in the `InterceptorsPreviewNamespaces` list roughly corresponds to one source generator. Well-behaved components are expected to not insert interceptors into namespaces they do not own.
```

Comment:
this does have some impact re accessibility, for nested types; I believe previously we could use `partial` to add intercepts inside the appropriate type to have access, but if we need to add the generators in a pre-advertised namespace, we can't do this any more; I guess we can emit a diagnostic to say "hey, to do the magic, I need that to be `internal` (if `private`) / `protected internal` (if `protected`)